optuna/artifacts/_backoff.py

```diff
@@ -112,6 +112,6 @@ def remove(self, artifact_id: str) -> None:
 if TYPE_CHECKING:
     # A mypy-runtime assertion to ensure that the Backoff middleware implements
     # all abstract methods in ArtifactStore.
-    from . import FileSystemArtifactStore
+    from optuna.artifacts._filesystem import FileSystemArtifactStore
```

Comment:
I think `FileSystemArtifactStore` is imported from `optuna/artifacts/__init__.py` in the original code. Do we need to import it from the private module (i.e., `_filesystem.py`)?
 
```suggestion
    from optuna.artifacts import FileSystemArtifactStore
```

Reply:
Also think it's unnecessary, so I fixed it. Then, how about `from optuna.artifacts._protocol import ArtifactStore`? Do you think it needs to be changed too?

Reply:
Thanks for your reply.

> How about `from optuna.artifacts._protocol import ArtifactStore`? 

I guess we can keep it since `ArtifactStore` is not exposed in `optuna/artifacts/__init__.py`.

Reply:
Ok, thanks for reply.